lib/collection/src/operations/verification/mod.rs

```diff
@@ -158,6 +163,33 @@ pub trait StrictModeVerification {
     }
 }
 
+fn check_filter_limits(
+    filter: &Filter,
+    strict_mode_config: &StrictModeConfig,
+) -> Result<(), CollectionError> {
+    if !check_custom(
+        || Some(filter.total_conditions_count()),
+        strict_mode_config.filter_max_conditions,
+    ) {
+        return Err(CollectionError::strict_mode(
+            "Filter condition limit reached",
+            "Reduce the amount of conditions of your filter.",
```

Comment:
I think the error reporting might frustrating for users.

They are told to reduce the number of conditions but there is no exact number to target.

Would it be difficult to return something clearer here?